modules/remotecontrol/Receiver.js

```diff
@@ -125,52 +156,123 @@ export default class Receiver extends RemoteControlParticipant {
-                APP.conference.screenSharingPromise.then(() => {
-                    if(APP.conference.isSharingScreen) {
-                        this._sendRemoteControlEvent(userId, {
-                            type: EVENT_TYPES.permissions,
-                            action: action
-                        });
-                    } else {
-                        this._sendRemoteControlEvent(userId, {
-                            type: EVENT_TYPES.permissions,
-                            action: PERMISSIONS_ACTIONS.error
-                        });
-                    }
-                }).catch(() => {
+        switch(action) {
+            case PERMISSIONS_ACTIONS.grant:
+                this.grant(userId);
+                break;
+            case PERMISSIONS_ACTIONS.deny:
+                this.deny(userId);
+                break;
+            case PERMISSIONS_ACTIONS.error:
+                this._sendRemoteControlEvent(userId, {
+                    type: EVENT_TYPES.permissions,
+                    action: action
+                });
+                break;
+            default:
+                // Unknown action. Ignore.
+        }
+    }
+
+    /**
+     * Denys remote control access for user associated with the passed user id.
+     *
+     * @param {string} userId - The id associated with the user who sent the
+     * request for remote control authorization.
+     * @returns {void}
+     */
+    deny(userId) {
+        this._sendRemoteControlEvent(userId, {
+            type: EVENT_TYPES.permissions,
+            action: PERMISSIONS_ACTIONS.deny
+        });
+    }
+
+    /**
+     * Grants remote control access to user associated with the passed user id.
+     *
+     * @param {string} userId - The id associated with the user who sent the
+     * request for remote control authorization.
+     * @returns {void}
+     */
+    grant(userId) {
+        APP.conference.addConferenceListener(ConferenceEvents.USER_LEFT,
+            this._userLeftListener);
+        this.controller = userId;
+        logger.log("Remote control permissions granted to: " + userId);
+        if(!APP.conference.isSharingScreen) {
+            APP.conference.toggleScreenSharing();
+            APP.conference.screenSharingPromise.then(() => {
+                if(APP.conference.isSharingScreen) {
+                    this._sendRemoteControlEvent(userId, {
+                        type: EVENT_TYPES.permissions,
+                        action: PERMISSIONS_ACTIONS.grant
+                    });
+                } else {
                     this._sendRemoteControlEvent(userId, {
                         type: EVENT_TYPES.permissions,
                         action: PERMISSIONS_ACTIONS.error
                     });
+                }
+            }).catch(() => {
+                this._sendRemoteControlEvent(userId, {
+                    type: EVENT_TYPES.permissions,
+                    action: PERMISSIONS_ACTIONS.error
```

Comment:
One case you might want to look out for now is there is no clearRequest call in controller on permission error so the UI looks like the remote control request is pending.